NineChronicles/2023-10/PlanetSpec.schema.json

```
@@ -0,0 +1,72 @@
+{    
+    "$id": "https://planetarium.github.io/json-schema/NineChronicles/2023-10/PlanetSpec.schema.json",
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "title": "PlanetSpec",
+    "type": "object",
+    "required": [
+        "id",
+        "name",
+        "rpcEndpoints"
+    ],
+    "properties": {
+        "id": {
+            "description": "The unique identifier for the planet",
+            "type": "string",
+            "pattern": "^0x[a-fA-F0-9]{12}$"
+        },
+        "name": {
+            "description": "The human-readable name for the planet",
+            "type": "string"
+        },
+        "genesisHash": {
+            "description": "The block hash of the genesis block for the planet",
+            "type": "string",
+            "pattern": "^[a-fA-F0-9]{64}$"
+        },
+        "genesisUri": {
+            "description": "The URI for fetching the genesis block content",
+            "type": "string",
+            "format": "uri"
+        },
+        "rpcEndpoints": {
+            "description": "The list of rpc endpoints for the planet",
+            "type": "object",
+            "required": [
+                "headless.gql",
+                "headless.grpc"
+            ],
+            "properties": {
+                "headless.gql": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "format": "uri"
+                    }
+                },
+                "headless.grpc": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "format": "uri"
+                    }
+                },
+                "dp.gql": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "format": "uri"
+                    }
+                },
+                "9cscan.rest": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "format": "uri"
+                    }
+                }
+            },
+            "additionalProperties": false
+        }
+    },
```

```suggestion
        "rpcEndpoints": {
            "description": "The list of rpc endpoints for the planet",
            "type": "array",
            "items": {
                "type": "object",
                "required": ["type", "uris"],
                "properties": {
                    "type": {
                        "type": "string",
                        "enum": ["headless.gql", "headless.grpc", "dp.gql", "9cscan.rest"]
                    },
                    "uris": {
                        "type": "array",
                        "items": {
                            "type": "string",
                            "format": "uri"
                        }
                    }
                }
            }
        },
```

What I thought was, maybe like this? 
Requiring "headless.gql", "headless.grpc" in schema level is complicated with this structure. also don't like the enum though.